### ISSUE TITLE
Traceback throws error in linprog

### ISSUE THREAD

**Issue by JayMan91** [scipy/scipy#10400]
My issue is about  callbacks on scipy.optimize.linprog 

### Reproducing code example:

```
import scipy
from scipy.optimize import linprog
c = [-1, 4]
A = [[-3, 1], [1, 2]]
b = [6, 4]
x0_bounds = (None, None)
x1_bounds = (-3, None)

def linprog_verbose_callback(res): # copied from this github repository
    """
    A sample callback function demonstrating the linprog callback interface.
    This callback produces detailed output to sys.stdout before each iteration
    and after the final iteration of the simplex algorithm.
    Parameters
    ----------
    res : A `scipy.optimize.OptimizeResult` consisting of the following fields:
        x : 1D array
            The independent variable vector which optimizes the linear
            programming problem.
        fun : float
            Value of the objective function.
        success : bool
            True if the algorithm succeeded in finding an optimal solution.
        slack : 1D array
            The values of the slack variables. Each slack variable corresponds
            to an inequality constraint. If the slack is zero, then the
            corresponding constraint is active.
        con : 1D array
            The (nominally zero) residuals of the equality constraints, that is,
            ``b - A_eq @ x``
        phase : int
            The phase of the optimization being executed. In phase 1 a basic
            feasible solution is sought and the T has an additional row
            representing an alternate objective function.
        status : int
            An integer representing the exit status of the optimization::
                 0 : Optimization terminated successfully
                 1 : Iteration limit reached
                 2 : Problem appears to be infeasible
                 3 : Problem appears to be unbounded
                 4 : Serious numerical difficulties encountered
        nit : int
            The number of iterations performed.
        message : str
            A string descriptor of the exit status of the optimization.
    """
    x = res['x']
    fun = res['fun']
    success = res['success']
    phase = res['phase']
    status = res['status']
    nit = res['nit']
    message = res['message']
    complete = res['complete']

    saved_printoptions = np.get_printoptions()
    np.set_printoptions(linewidth=500,
                        formatter={'float': lambda x: "{0: 12.4f}".format(x)})
    if status:
        print('--------- Simplex Early Exit -------\n'.format(nit))
        print('The simplex method exited early with status {0:d}'.format(status))
        print(message)
    elif complete:
        print('--------- Simplex Complete --------\n')
        print('Iterations required: {}'.format(nit))
    else:
        print('--------- Iteration {0:d}  ---------\n'.format(nit))

    if nit > 0:
        if phase == 1:
            print('Current Pseudo-Objective Value:')
        else:
            print('Current Objective Value:')
        print('f = ', fun)
        print()
        print('Current Solution Vector:')
        print('x = ', x)
        print()

    np.set_printoptions(**saved_printoptions)


res = linprog(c, A_ub=A, b_ub=b, bounds=[x0_bounds, x1_bounds],callback=linprog_verbose_callback)
```

### Error message:

```
IndexError                                Traceback (most recent call last)
<ipython-input-29-370a3c2dde72> in <module>
      6 x1_bounds = (-3, None)
      7 from scipy.optimize import linprog
----> 8 res = linprog(c, A_ub=A, b_ub=b, bounds=[x0_bounds, x1_bounds],callback=linprog_verbose_callback)

~/anaconda3/lib/python3.7/site-packages/scipy/optimize/_linprog.py in linprog(c, A_ub, b_ub, A_eq, b_eq, bounds, method, callback, options)
    464         if meth == 'simplex':
    465             x, status, message, iteration = _linprog_simplex(
--> 466                 c, c0=c0, A=A, b=b, callback=callback, _T_o=T_o, **solver_options)
    467         elif meth == 'interior-point':
    468             x, status, message, iteration = _linprog_ip(

~/anaconda3/lib/python3.7/site-packages/scipy/optimize/_linprog_simplex.py in _linprog_simplex(c, c0, A, b, maxiter, disp, callback, tol, bland, _T_o, **unknown_options)
    581 
    582     nit1, status = _solve_simplex(T, n, basis, phase=1, callback=callback,
--> 583                                   maxiter=maxiter, tol=tol, bland=bland, _T_o=_T_o)
    584     # if pseudo objective is zero, remove the last row from the tableau and
    585     # proceed to phase 2

~/anaconda3/lib/python3.7/site-packages/scipy/optimize/_linprog_simplex.py in _solve_simplex(T, n, basis, maxiter, phase, status, message, callback, tol, nit0, bland, _T_o)
    375             c, A_ub, b_ub, A_eq, b_eq, bounds, undo = _T_o
    376             x, fun, slack, con, _, _ = _postsolve(
--> 377                 x, c, A_ub, b_ub, A_eq, b_eq, bounds, undo=undo, tol=tol
    378             )
    379             res = OptimizeResult({

~/anaconda3/lib/python3.7/site-packages/scipy/optimize/_linprog_util.py in _postsolve(x, c, A_ub, b_ub, A_eq, b_eq, bounds, complete, undo, tol)
   1115             if lb is None and ub is None:
   1116                 n_unbounded += 1
-> 1117                 x[i] = x[i] - x[n_x + n_unbounded - 1]
   1118             else:
   1119                 if lb is None:

IndexError: index 2 is out of bounds for axis 0 with size 2
  ...
```

### Scipy/Numpy/Python version information:
1.2.1 1.16.4 sys.version_info(major=3, minor=7, micro=3, releaselevel='final', serial=0)

**Comment by mdhaber** [comment]
Thanks for your patience. I can't look into this in detail right now, but can you upgrade to SciPy 1.3 and see if that fixes your issue?
You might also try using `method='interior-point'` in 1.2.1, or if you upgrade,  `method='revised simplex'`.

**Comment by Kai-Striega** [comment]
@JayMan91 thanks for raising this as an issue. [pr-9563](https://github.com/scipy/scipy/pull/9563) fixed this in SciPy 1.3 and it should now work as expected.

**Comment by JayMan91** [comment]
Thanks a lot, I am going to try after upgrading.
